tensorizer/stream_io.py

```diff
@@ -897,23 +908,24 @@ def _ensure_https_endpoint(endpoint: str):
         raise ValueError("Non-HTTPS endpoint URLs are not allowed.")
 
 
+def _is_caios(endpoint: str) -> bool:
+    host = urlparse(endpoint if "//" in endpoint else "//" + endpoint).netloc
+    return host.lower() in {"cwobject.com", "cwlota.com"}
```

Comment:
Really not a fan of us hardcoding these like this (or the ord1 defaults we have had elsewhere for some time). But can be convinced this is ok.

For reference to those other hardcoded values that should be either A) dropped, or B) changed to our current object storage offering.
``` python
default_s3_read_endpoint = "accel-object.ord1.coreweave.com"
default_s3_write_endpoint = "object.ord1.coreweave.com"
```

Reply:
I'm pretty sure in `boto3`'s source tree there is a large data segment that has these sorts of hardcoded compatibility checks for a huge set of different AWS endpoints, which is how Boto3's "magically pick the right one" feature works. Unless/until we upstream a change to support our object storage endpoints correctly, which I'm not sure they'd accept, this seems to me like a fair place to put a workaround.

The other hardcoded values are necessary for the default `s3://tensorized` bucket to "just work" since they're in that location. I'd like to change it to the newer endpoint in release 3.0.

Reply:
If this is a pattern already used in boto3 then probably fine... although we are not automatically changing the `force_http` based on if we are using `cwlota.com` or not which would fall into the same pattern as what we are doing here.  Same points apply on both sides there as well as if a user configures it correctly it will work, but by default with a minimal amount of configuring it will not.

Reply:
I am nervous about defaulting `force_http=True` for the `cwlota.com` endpoint on the basis that it could transmit a presigned URL as part of a request over an unknown network in cleartext if it is used somewhere where `cwlota.com` doesn't resolve to a local network (so not a CW datacenter, maybe from someone's own machine where they're trying to test something), although I think the presigned URL would only be valid for use by someone who can access `cwlota.com` endpoints. I'd rather leave that one specifically as something that needs to be declared explicitly in the config.